src/renderer/components/status-bar/status-bar-item-registrator.injectable.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import type { Injectable } from "@ogre-tools/injectable";
+import { getInjectable } from "@ogre-tools/injectable";
+import { computed } from "mobx";
+import { extensionRegistratorInjectionToken } from "../../../extensions/extension-loader/extension-registrator-injection-token";
+import type { LensRendererExtension } from "../../../extensions/lens-renderer-extension";
+import type { StatusBarItem } from "./status-bar-item-injection-token";
+import { statusBarItemInjectionToken } from "./status-bar-item-injection-token";
+import type { StatusBarRegistration } from "./status-bar-registration";
+import React from "react";
+import getRandomIdInjectable from "../../../common/utils/get-random-id.injectable";
+import logger from "../../../common/logger";
+
+const statusBarItemRegistratorInjectable = getInjectable({
+  id: "status-bar-item-registrator",
+
+  instantiate: (di) => (extension) => {
+    const rendererExtension = extension as LensRendererExtension;
+    const getRandomId = di.inject(getRandomIdInjectable);
+
+    return rendererExtension.statusBarItems.flatMap(
+      toItemInjectableFor(rendererExtension, getRandomId),
+    );
+  },
+
+  injectionToken: extensionRegistratorInjectionToken,
+});
+
+export default statusBarItemRegistratorInjectable;
+
+const toItemInjectableFor = (extension: LensRendererExtension, getRandomId: () => string) => {
+  return (registration: StatusBarRegistration): Injectable<StatusBarItem, StatusBarItem, void>[] => {
+    const id = `${getRandomId()}-status-bar-item-for-extension-${extension.sanitizedExtensionId}`;
+    let component: React.ComponentType;
```

Comment:
nit: kinda hard to follow since we are using variables which are set in different timeslots. Could be simplified.